pandas/indexes/multi.py

```diff
@@ -490,6 +491,11 @@ def _set_names(self, names, level=None, validate=True):
         that it only acts on copies
         """
 
+        # GH 15110
+        # Don't allow a single string for names in a MultiIndex
+        if names is not None and is_string_like(names):
```

Comment:
I would use ``not is_list_like`` here (which excludes things like TImestamps and strings), but allows other list-like things